libs/datasets/timeseries.py

```diff
@@ -952,3 +953,97 @@ def drop_regions_without_population(
             "Dropping unexpected regions without populaton", location_ids=sorted(unexpected_drops)
         )
     return mrts.get_locations_subset(locations_with_population)
+
+
+LOCATION_ID_AGG = "location_id_agg"
+
+
+def _aggregate_dataframe_by_region(
+    df_in: pd.DataFrame, location_id_map: Mapping[str, str]
+) -> pd.DataFrame:
```

Comment:
The logic here overall seems correct, but I'm a bit unsure it'll provide for easy flexibility with some of the other sorts of calculations.

How would you feel about an approach that takes advantage of the apply function with a named series? I would imagine it would be something like:

```python
def sum_no_nan(series):
    return series.sum(min_count=len(series))

data.groupby(groupby_columns).apply(
    lambda x: pd.Series({
        column: sum_no_nan(x[column])
        for column in columns
    })
)
```

The idea being that it would be pretty easy to treat different columns differently and apply different aggregating functions. 

It then lessens the need to go back and forth between wide and long form columns and I think the code will be a bit more concise to digest.

Reply:
A simple test found this will take ~30min to run with bare python, vs seconds without apply. It is quite possible numba can bring that down to a few seconds, but I haven't tried.  See also the description of https://github.com/covid-projections/covid-data-model/pull/813